asmcomp/cmm_helpers.ml

```diff
@@ -2396,14 +2396,15 @@ let bytesset_unsafe arg1 arg2 arg3 dbg =
 
 let bytesset_safe arg1 arg2 arg3 dbg =
   return_unit dbg
-    (bind "str" arg1 (fun str ->
+    (bind "newval" (untag_int arg3 dbg) (fun newval ->
       bind "index" (untag_int arg2 dbg) (fun idx ->
+       bind "str" arg1 (fun str ->
         Csequence(
           make_checkbound dbg [string_length str dbg; idx],
           Cop(Cstore (Byte_unsigned, Assignment),
               [add_int str idx dbg;
-               ignore_high_bit_int (untag_int arg3 dbg)],
-              dbg)))))
+               ignore_high_bit_int newval],
```

Comment:
I'd rather leave the `ignore_high_bit_int` next to the `untag_int`, otherwise some optimizations may not apply.  Something like
```
  (bind "newval" (ignore_high_bit_int (untag_int arg3 dbg)) (fun newval -> ...
```

Reply:
Thank you for the review.  I fixed the placement of `ignore_high_bit_int`, rebased this PR, and the CI is green.